polyfill/lib/ecmascript.mjs

```diff
@@ -282,32 +284,33 @@ export const ES = ObjectAssign({}, ES2019, {
     return { hour, minute, second, millisecond, microsecond, nanosecond };
   },
   RegulateYearMonth: (year, month, disambiguation) => {
+    const refISODay = 1;
     switch (disambiguation) {
       case 'reject':
-        ES.RejectYearMonth(year, month);
+        ES.RejectYearMonth(year, month, refISODay);
         break;
       case 'constrain':
         ({ year, month } = ES.ConstrainYearMonth(year, month));
         break;
       case 'balance':
         ({ year, month } = ES.BalanceYearMonth(year, month));
         // Still rejected if balanced YearMonth is outside valid range
-        ES.RejectYearMonth(year, month);
+        ES.RejectYearMonth(year, month, refISODay);
```

Comment:
We can't, since the range of YearMonth is slightly longer than that of Date:
```js
Temporal.Date.from('-271821-04-19').getYearMonth().getISOCalendarFields()
  // => { year: -271821, month: 4, refISODay: 1 }
```
Or, the alternative is that we could use RejectDate but the ISO calendar would have to set the refISODay for that particular month to 1 so that it fit inside the Temporal.Date range.

Reply:
No strong opinion here, except to note that it would be weird if refISODay were *not* 1 for the ISO calendar.

Reply:
Yeah, I mistyped above ... I meant "set the refISODay to _something other than_ 1" which I think agrees with what you said.

Reply:
This does seem like an undesirable papercut.  Can we make Temporal.Date and Temporal.DateTime end at a year boundary?  Or is the limit chosen for legacy Date compatibility or something like that?

Reply:
I'm not sure I see what the papercut is, since nothing has changed for API users; the principle is still:
- Temporal.Absolute supports the full range supported by legacy Date
- Temporal.DateTime's range is enough that `absolute.inTimeZone()` will never be out of range
- Temporal.Date's range is enough that `dateTime.getDate()` will never be out of range
- Temporal.YearMonth's range is enough that `date.getYearMonth()` will never be out of range

Specifically, this does not clip Temporal.YearMonth's range to Temporal.Date's range, whereas changing the spec internally to use RejectDate instead of RejectYearMonth would clip.

Reply:
The papercut is that `refISODay` is not unconditionally 1.

It seems that we already have the problem that `dateTime.inTimeZone()` could return an out-of-range Absolute.  Can we therefore just extend the range of DateTime to the nearest month boundary?

Reply:
I see the disconnect now — that papercut was just hypothetical. We do actually have `refISODay` always equal to 1 in the ISO calendar. Maintaining that would have been incompatible with with the suggestion of using RejectDate instead of RejectYearMonth, but I didn't adopt that suggestion. So if I understand correctly, no change is needed.

Reply:
Sounds good.